core/src/replay_stage.rs

```diff
@@ -1349,14 +1349,18 @@ impl ReplayStage {
                         );
                     }
 
-
                     // Should not dump slots for which we were the leader
                     if Some(*my_pubkey) == leader_schedule_cache.slot_leader_at(*duplicate_slot, None) {
-                            panic!("We are attempting to dump a block that we produced. \
-                                This indicates that we are producing duplicate blocks, \
-                                or that there is a bug in our runtime/replay code which \
-                                causes us to compute different bank hashes than the rest of the cluster. \
-                                We froze slot {duplicate_slot} with hash {frozen_hash:?} while the cluster hash is {correct_hash}");
+                        if let Some(duplicate_bank) = bank_forks.read().unwrap().get(*duplicate_slot) {
+                            let _ = bank_hash_details::write_bank_hash_details_file(&duplicate_bank);
```

Comment:
nit: instead of ignoring the `Result` should we log the details of why we can't write the bank hash details?

Reply:
Good call - done in https://github.com/solana-labs/solana/pull/34256/commits/4ef6b525f88715be19f71a2cf7c4e48050adb419